### DISCUSSION TITLE
Is it possible to update cache from urql client?

### DISCUSSION THREAD

**Discussion by nghiepdev** [urql-graphql/urql#1894] [Q&A] [answered]
Hi,

Is there a way can I update the cache from the client?
```jsx
import {useClient} from 'urql';

const client = useClient();
const data = await fetch('/api/from-rest');
client.setCache(MY_QUERY, data);
```
Or, `client.readQuery` should be support for fallback function.
```jsx
const client = useClient();
client.readQuery(MY_QUERY, {}, {
  fallbackResolve() {
    return fetch('/api/from-rest');
  }
})
```

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

We're knowingly not allowing for this as we've always aimed to be a presentation of your server-side state, there's a thread about allowing direct access to graphCache for similar reasons [here](https://github.com/FormidableLabs/urql/issues/1725) which elaborates on more of that reasoning.

**Reply by nghiepdev** [reply]
Thanks @JoviDeCroock
